ClientRuntimes/NodeJS/ms-rest-azure/lib/azureEnvironment.js

```diff
@@ -2,48 +2,158 @@
+    galleryEndpointUrl: 'https://gallery.chinacloudapi.cn/',
+    activeDirectoryEndpointUrl: 'https://login.chinacloudapi.cn',
+    activeDirectoryResourceId: 'https://management.core.chinacloudapi.cn/',
+    activeDirectoryGraphResourceId: 'https://graph.chinacloudapi.cn/',
+    activeDirectoryGraphApiVersion: '2013-04-05',
+    storageEndpointSuffix: '.core.chinacloudapi.cn',
+    keyVaultDnsSuffix: '.vault.azure.cn',
+    // TODO: add dns suffixes for the china cloud for datalake store and datalake analytics once they are defined.
+    azureDataLakeStoreFileSystemEndpointSuffix: 'N/A',
+    azureDataLakeAnalyticsCatalogAndJobEndpointSuffix: 'N/A'
+  },
+  AzureUSGovernment: {
+    name: 'AzureUSGovernment',
+    portalUrl: 'https://manage.windowsazure.us',
+    publishingProfileUrl: 'https://manage.windowsazure.us/publishsettings/index',
+    managementEndpointUrl: 'https://management.core.usgovcloudapi.net',
+    resourceManagerEndpointUrl: 'https://management.usgovcloudapi.net',
+    sqlManagementEndpointUrl: 'https://management.core.usgovcloudapi.net:8443/',
+    sqlServerHostnameSuffix: '.database.usgovcloudapi.net',
+    galleryEndpointUrl: 'https://gallery.usgovcloudapi.net/',
+    activeDirectoryEndpointUrl: 'https://login.microsoftonline.com',
+    activeDirectoryResourceId: 'https://management.core.usgovcloudapi.net/',
+    activeDirectoryGraphResourceId: 'https://graph.windows.net/',
+    activeDirectoryGraphApiVersion: '2013-04-05',
+    storageEndpointSuffix: '.core.usgovcloudapi.net',
+    keyVaultDnsSuffix: '.vault.usgovcloudapi.net',
+    // TODO: add dns suffixes for the US government for datalake store and datalake analytics once they are defined.
+    azureDataLakeStoreFileSystemEndpointSuffix: 'N/A',
+    azureDataLakeAnalyticsCatalogAndJobEndpointSuffix: 'N/A'
+  },
+  AzureGermanCloud: {
+    name: 'AzureGermanCloud',
+    portalUrl: 'http://portal.microsoftazure.de/',
+    publishingProfileUrl: 'https://manage.microsoftazure.de/publishsettings/index',
+    managementEndpointUrl: 'https://management.core.cloudapi.de',
+    resourceManagerEndpointUrl: 'https://management.microsoftazure.de',
+    sqlManagementEndpointUrl: 'https://management.core.cloudapi.de:8443/',
+    sqlServerHostnameSuffix: '.database.cloudapi.de',
+    galleryEndpointUrl: 'https://gallery.cloudapi.de/',
+    activeDirectoryEndpointUrl: 'https://login.microsoftonline.de',
+    activeDirectoryResourceId: 'https://management.core.cloudapi.de/',
+    activeDirectoryGraphResourceId: 'https://graph.cloudapi.de/',
+    activeDirectoryGraphApiVersion: '2013-04-05',
+    storageEndpointSuffix: '.core.cloudapi.de',
+    keyVaultDnsSuffix: '.vault.microsoftazure.de',
+    // TODO: add dns suffixes for the US government for datalake store and datalake analytics once they are defined.
+    azureDataLakeStoreFileSystemEndpointSuffix: 'N/A',
+    azureDataLakeAnalyticsCatalogAndJobEndpointSuffix: 'N/A'
+  }
+};
```

Comment:
Dont want to expose the internal environments here. Have provided a mechanism to add new environment.